tests/test_runner/test_discover_runner.py

```diff
@@ -259,6 +259,11 @@ def count_tests(**kwargs):
         self.assertEqual(count_tests(tags=['foo'], exclude_tags=['bar', 'baz']), 1)
         self.assertEqual(count_tests(exclude_tags=['foo']), 0)
 
+    def test_tag_failed_tests(self):
+        runner = DiscoverRunner(tags=['foo'])
+        # The "doesnotexist" label results in a _FailedTest instance.
+        self.assertEqual(runner.build_suite(['doesnotexist']).countTestCases(), 1)
```

Comment:
I think tests with syntax errors are more important for this issue, so I added a test for them.

Reply:
Great, thanks! However, are you sure you're testing `SyntaxError` with your new test module? It looks like `NameError`. By using a matching tag (`syntax_error`), the test will still pass with `NameError`. It would be better to (1) use a non-matching tag, and (2) perhaps test that it's really a `SyntaxError`.

Reply:
Changing the last line to `syntax_error:  # NOQA` (with colon) or `a syntax_error  # NOQA` (two words) are a couple ways to do it.

Reply:
> By using a matching tag (`syntax_error`), the test will still pass with `NameError`. It would be better to (1) use a non-matching tag, and ...

I wanted a regression test for the scenario described in the ticket.

> ... (2) perhaps test that it's really a `SyntaxError`.

Thanks. I've changed to a `SyntaxError`.